### PR TITLE
refactor: adopt parcel toolchain

### PR DESCRIPTION
**Pull request by LeoDog896**
This uses the [Parcel](https://parceljs.org/) toolchain, with the following refactrs:
- Offsets copying the book to parcel
- Moves the favicon images to one, resizable image
- Uses Split.js's [npm module](https://www.npmjs.com/package/split.js) instead of browser script

Current known issues with a full move to parcel (thus why partial codemirror src && some image assets are still there):
- Codemirror v5's npm module just acts *weird*, and doesn't work even if you import it without parcel's bundling feature. It's most likely necessary to make another PR that refactors to codemirror v6.
- when using [query resizing](https://parceljs.org/recipes/image/#resizing-and-converting-images) (`?width=X&height=Y`), mstile causes duplicate assets in parcel (https://github.com/parcel-bundler/parcel/issues/8821#issuecomment-1423195983) meaning that not all of the mstile images could be moved (excluding the 310x150 long image)

~~Note: The CI fails in this PR as I've added an extra action to install PNPM.~~ 

I've mostly avoided any script refactors as requested (#21), and will wait to the next PR to refactor scripts.

This also fixes the error that occurred in #24 by installing necessary dependencies using pnpm (`@parcel/config-default` wasn't being installed)

### CODE REVIEW

**Comment by LeoDog896** [comment]
*I've marked it as a draft so I can make sure that the CI works by changing it to run push on all branches.

**Comment by huacnlee** [comment]
I am strong to recommend we use Vite.

Vite is a next generation frontend tool.
It's simple and fast. Vite builds as [ES modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) for modern browsers.

https://npmtrends.com/parcel-vs-snowpack-vs-vite

**Comment by nathanielknight** [comment]
If we're going by popularity [we should clearly use `esbuild`](https://npmtrends.com/esbuild-vs-parcel-vs-snowpack-vs-vite) 😜 

I think I prefer this PR to #33; the more tools we adopt, the more things a potential contributor has to know. Grabbing too many tools means we might have trouble finding folks to work on the site in the future, and might be discouraging for folks who aren't as familiar with web technology. I think adopting Parcel or Vite or ESBuild to help us organize the code is probably worthwhile, but I think adopting Tailwind and React might lead to problems down the line.

**Comment by LeoDog896** [comment]
Agreed - I am also not a big fan of using tailwind for css in open source sites.

**Comment by huacnlee** [comment]
I understand what you mean and agree with it.
Actually, Parcel or Vite is acceptable to me (much better than none at the moment).

> I actually didn't see your discussion until I implemented #33.

So if you both agree that Parcel is your first choice, then make that decision, Parcel is ok for me, and don’t need to care about my PR.

The current problem facing the Pest site is the need to introduce a frontend tool as soon as possible, and refactor the inline JavaScript implementation (use Npm to manage three-party libraries, and independent JS files, introduce TypeScript and ES6 syntax), so as to create more rich features are possible.

That's why I suggest introducing Vite and TypeScript or even React (when I need to implement Playground, these modern tools will be able to build better complex and complete functional details). Tailwind CSS just a recommendation, I don't particularly care whether must to use it or not.

> 💡 There is a complete TypeScript implementation extracted from HTML in PR #33. If you think this needs to be done, @LeoDog896 can extract it directly from there and quickly promote the Pest site to complete these transformations. Don't mind the contributor info, I don't mind.

In fact, I have observed that the introduction of Parcel seems to have been a long time ago. Until I added the Format function yesterday, the Pest site still did not have an efficient front-end packaging tool.

So my suggestion is to complete the merge of this PR as soon as possible. If problems with such tools are found later, we can still improve quickly (actually, do like this things just need about 1-2 hours).

**Comment by huacnlee** [comment]
CI need to add:

```yml

jobs:
  checks:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: pnpm/action-setup@v2
        with:
          version: 6.0.2
```

https://github.com/pnpm/action-setup

**Review by nathanielknight — APPROVED** [review]

**Comment by nathanielknight** [comment]
I'm good with adopting Parcel quickly and then focusing on some more improvements (I think migrating to TS before we add more code is a great idea, and it looks like [Parcel transpiles it automatically](https://parceljs.org/languages/typescript).

**Comment by LeoDog896** [comment]
Alright, I'll fix the CI and set this as ready to merge. Any quirks in the CI we can try to fix it as we go, as github CI changes in PRs is pretty frusturating.

**Comment by LeoDog896** [comment]
> CI need to add:
> 
> ```yaml
> jobs:
>   checks:
>     runs-on: ubuntu-latest
>     steps:
>       - uses: actions/checkout@v2
>       - uses: pnpm/action-setup@v2
>         with:
>           version: 6.0.2
> ```
> 
> https://github.com/pnpm/action-setup

Whoops, almost missed this - thanks!

**Comment by huacnlee** [comment]
I have been running your updates in local, the webpage is looks ok.

You may need update README for new development guide.

````md
# Source code for `https://pest.rs`

This repo contains the source code for https://pest.rs

## Development Guide

This project uses the following tools:

- [Task](https://taskfile.dev/)
- [wasm-pack](https://rustwasm.github.io/wasm-pack/)
- [mdBook](https://rust-lang.github.io/mdBook/)
- [Pnpm](https://pnpm.io)

Install the dependencies first:

```bash
pnpm install
```

Build the WASM library:

```bash
task
```

Start Parcel development server:

```bash
pnpm start
```

Then visit http://localhost:1234
````

**Comment by LeoDog896** [comment]
There's a few horrors in the merge conflict that I can't do with GH's web editor (I'm currently on mobile as of now), so it'll be a bit before I'm able to resolve them.

**Review by nathanielknight — APPROVED** [review]

**Comment by LeoDog896** [comment]
Terribly sorry but I'll be out for a bit. Edits can be made by maintainers, so if you want to merge the PR to unblock #18 in the meantime, someone has to fix the merge conflict.

File: .github/workflows/publish.yml
Comment (tomtau) [inline .github/workflows/publish.yml:49]: isn't push event even when someone pushes on pull requests?

**Review by tomtau — COMMENTED** [review]
@nathanielknight @huacnlee will you be able to look into the merge conflict?

File: .github/workflows/publish.yml
Comment (LeoDog896) [reply .github/workflows/publish.yml:49]: Since we filter the push event to the `master` branch, not only does [push not run on PRs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push), but it should also only run in the master branch.

**Comment by LeoDog896** [comment]
Merge conflict has been resolved - though I do want to make sure that I didn't break any existing functionality. Otherwise, it's ready :+1:

**Comment by LeoDog896** [comment]
*whoops, almost broke formatting :sweat_smile:

**Comment by tomtau** [comment]
@LeoDog896 the deployment seems to work fine now👍 , but clicking on the "wide button" doesn't work

**Comment by LeoDog896** [comment]
> @LeoDog896 the deployment seems to work fine now+1 , but clicking on the "wide button" doesn't work

whoops! I'll make a PR to fix that functionality.

**Pull request merged**
